scrapers/nfl_fantasy.py

```diff
@@ -0,0 +1,101 @@
+import csv
+import requests
+from bs4 import BeautifulSoup as BS
+
+
+NFL_FAN_POSITIONS = ['QB', 'RB', 'WR', 'TE', 'DEF']
+
+NFL_FAN_URL = \
+    'http://fantasy.nfl.com/research/projections?offset={}&position={}'
+
+DEFAULT_FILE_PATH = 'data/current-nfl-fan-projections.csv'
+
+
+def scrape(limit=1000, file_path=DEFAULT_FILE_PATH):
+    hold = scrape_nfl_fan(limit)
+    write_to_csv(hold, file_path)
+
+
+def scrape_nfl_fan(limit):
+    hold = []
+    for i in range(0, limit/25):
+        param = (i * 25) + 1
+        r = requests.get(NFL_FAN_URL.format(param, 'O'))
+        soup = BS(r.text, 'html.parser')
+        player_rows = filter(lambda x: x,
+                             [x.find_all('td') for x in soup.find_all('tr')])
+        for row in player_rows:
+            player_stats = \
+                map(lambda x:
+                    unicode('0') if x == '-' else x, [x.text for x in row])
+            if any(map(lambda x: x in player_stats[0], NFL_FAN_POSITIONS)):
+                pos = filter(lambda x:
+                             x in player_stats[0], NFL_FAN_POSITIONS)[0]
+                name = player_stats[0].split(pos)[0].rstrip()
+                projected_points = calculate_ppr(pos, player_stats)
+                if projected_points != 0:
+                    hold.append([name, projected_points])
+    for i in (0, 25):
+        r = requests.get(NFL_FAN_URL.format(i, '8'))
+        soup = BS(r.text, 'html.parser')
+        player_rows = filter(lambda x: x,
+                             [x.find_all('td') for x in soup.find_all('tr')])
+        for row in player_rows:
+            player_stats = \
+                map(lambda x:
+                    unicode('0') if x == '-' else x, [x.text for x in row])
+            if any(map(lambda x: x in player_stats[0], ['DEF'])):
+                name = player_stats[0].split('DEF')[0].rstrip()
+                projected_points = calculate_ppr('DEF', player_stats)
+                hold.append([name, projected_points])
+    return hold
+
+
+def calculate_ppr(pos, row):
+    projected_points = 0
```

Comment:
You're going to hate me on this one, but if we want to do this for a bunch of sites that provide projections but not the DraftKings way, I recommend we have a general `calculate_ppr` method that takes an object as its argument whose properties are position, projected catches, projected yards, etc. This will also make updating stuff when the NFL / Fantasy Pros changes their UI way easier.

Also, imagine how much easier `6 * player_proj.touchdowns` is instead of `6 * float(row[6])`.